test/test-tarantool/call.result

```diff
@@ -0,0 +1,599 @@
+call  f1 ()
+- []
+call  f2 ()
+- [[]]
+function f3() return {{'hello'}, {'world'}} end
+---
+...
+call  f3 ()
+- [['hello'], ['world']]
+function f3() return {'hello', {'world'}} end
+---
+...
+call  f3 ()
+- ['hello', ['world']]
+function f3() return 'hello', {{'world'}, {'canada'}} end
+---
+...
+call  f3 ()
+- 'hello'
+- [['world'], ['canada']]
+function f3() return {}, '123', {{}, {}} end
+---
+...
+call  f3 ()
+- []
+- '123'
+- [[], []]
+function f3() return { {{'hello'}} } end
+---
+...
+call  f3 ()
+- [[['hello']]]
+function f3() return { box.tuple.new('hello'), {'world'} } end
+---
+...
+call  f3 ()
+- [['hello'], ['world']]
+function f3() return { {'world'}, box.tuple.new('hello') } end
+---
+...
+call  f3 ()
+- [['world'], ['hello']]
+function f3() return { { test={1,2,3} }, { test2={1,2,3} } } end
+---
+...
+call  f3 ()
+- [{'test': [1, 2, 3]}, {'test2': [1, 2, 3]}]
+call  f1 ('jason',)
+- ['jason']
+call  f1 ('jason', 1, 'test', 2, 'stewart')
+- ['jason', 1, 'test', 2, 'stewart']
+space = box.schema.space.create('tweedledum')
+---
+...
+index = space:create_index('primary', { type = 'hash' })
+---
+...
+function myreplace(...) return space:replace{...} end
+---
+...
+function myinsert(...) return space:insert{...} end
+---
+...
+call  myinsert (1, 'test box delete')
+- [1, 'test box delete']
+call  space:delete (1,)
+- [1, 'test box delete']
+call  myinsert (1, 'test box delete')
+- [1, 'test box delete']
+call  space:delete (1,)
+- [1, 'test box delete']
+call  space:delete (1,)
+
+call  myinsert (2, 'test box delete')
+- [2, 'test box delete']
+call  space:delete (1,)
+
+call  space:delete (2,)
+- [2, 'test box delete']
+call  space:delete (2,)
+
+space:delete{2}
+---
+...
+call  myinsert (2, 'test box delete')
+- [2, 'test box delete']
+call  space:get (2,)
+- [2, 'test box delete']
+space:delete{2}
+---
+- [2, 'test box delete']
+...
+call  space:get (2,)
+
+call  myinsert (2, 'test box.select()')
+- [2, 'test box.select()']
+call  space:get (2,)
+- [2, 'test box.select()']
+call  space:select (2,)
+- [[2, 'test box.select()']]
+space:get{2}
+---
+- [2, 'test box.select()']
+...
+space:select{2}
+---
+- - [2, 'test box.select()']
+...
+space:get{1}
+---
+...
+space:select{1}
+---
+- []
+...
+call  myreplace (2, 'hello', 'world')
+- [2, 'hello', 'world']
+call  myreplace (2, 'goodbye', 'universe')
+- [2, 'goodbye', 'universe']
+call  space:get (2,)
+- [2, 'goodbye', 'universe']
+call  space:select (2,)
+- [[2, 'goodbye', 'universe']]
+space:get{2}
+---
+- [2, 'goodbye', 'universe']
+...
+space:select{2}
+---
+- - [2, 'goodbye', 'universe']
+...
+call  myreplace (2,)
+- [2]
+call  space:get (2,)
+- [2]
+call  space:select (2,)
+- [[2]]
+call  space:delete (2,)
+- [2]
+call  space:delete (2,)
+
+call  myinsert (3, 'old', 2)
+- [3, 'old', 2]
+call  myinsert (3, 'old', 2)
+{
+    "error": {
+        "code": "ER_TUPLE_FOUND", 
+        "reason": "Duplicate key exists in unique index 'primary' in space 'tweedledum'"
+    }
+}
```

Comment:
It gives the following miscompare on a recent tarantool version:

```diff
--- test/test-tarantool/call.result	2021-04-30 01:26:24.708845662 +0300
+++ test/var/rejects/test-tarantool/call.reject	2021-04-30 02:21:26.230637668 +0300
@@ -251,7 +251,7 @@
 {
     "error": {
         "code": "ER_TUPLE_FOUND", 
-        "reason": "Duplicate key exists in unique index 'primary' in space 'tweedledum'"
+        "reason": "Duplicate key exists in unique index \"primary\" in space \"tweedledum\" with old tuple - [3, \"old\", 2] and new tuple - [3, \"old\", 2]"
     }
 }
 space:update({3}, {{'=', 1, 4}, {'=', 2, 'new'}})
```

See https://github.com/tarantool/tarantool/commit/d11fb3061e15faf4e0eb5375fb8056b4e64348ae

You can use a 'push filter' / 'clear filter' test-run commands to preprocess the message (or do it manually in the Lua code, or just remove the field if you don't need it for testing of test-run).

Reply:
removed 
```
--- a/test/test-tarantool/call.result
+++ b/test/test-tarantool/call.result
@@ -247,13 +247,6 @@ call  space:delete (2,)
 
 call  myinsert (3, 'old', 2)
 - [3, 'old', 2]
-call  myinsert (3, 'old', 2)
-{
-    "error": {
-        "code": "ER_TUPLE_FOUND", 
-        "reason": "Duplicate key exists in unique index 'primary' in space 'tweedledum'"
-    }
-}
 space:update({3}, {{'=', 1, 4}, {'=', 2, 'new'}})
 ---
 - error: Attempt to modify a tuple field which is part of index 'primary' in space
```